src/Microsoft.ML.Transforms/PermutationFeatureImportanceExtensions.cs

```diff
@@ -348,6 +648,101 @@ private static double[] ComputeSequenceDeltas(IReadOnlyList<double> a, IReadOnly
             return delta;
         }
 
+        private static ImmutableDictionary<string, TResult>
+            PermutationFeatureImportance<TMetric, TResult>(
+                IHostEnvironment env,
+                ITransformer model,
+                IDataView data,
+                Func<TResult> resultInitializerParam,
+                Func<TMetric, TMetric, TMetric> deltaFuncParam,
+                Func<IDataView, TMetric> evaluationFuncParam,
+                bool useFeatureWeightFilter,
+                int? numberOfExamplesToUse,
+                int permutationCount) where TResult : IMetricsStatistics<TMetric>
+        {
+            env.CheckValue(data, nameof(data));
+            env.CheckValue(model, nameof(model));
+
+            ITransformer lastTransformer = null;
+
+            if (model is TransformerChain<ITransformer> chain)
+            {
+                foreach (var transformer in chain.Reverse())
+                {
+                    if (transformer is ISingleFeaturePredictionTransformer)
+                    {
+                        lastTransformer = transformer;
+                        break;
+                    }
+                }
+            }
+            else lastTransformer = model as ISingleFeaturePredictionTransformer;
+
+            env.CheckValue(lastTransformer, nameof(lastTransformer));
+
+            var lastTransformerType = lastTransformer.GetType();
+            string featureColumnName = (lastTransformer as ISingleFeaturePredictionTransformer).FeatureColumnName;
+            Type pfiType = typeof(PermutationFeatureImportance<,,>);
+
+            TryGetImplementedIPredictionTransformer(lastTransformerType, out lastTransformerType);
```

Comment:
What happens if `TryGetImplementedIPredictionTransformer` returns `false`?

Reply:
I don't think that case can ever happen.

`ISingleFeaturePredictionTransformer<>` extends `IPredictionTransformer<>`, which is what we are checking for here. If something isn't an `ISingleFeaturePredictionTransformer<>`, it should fail the check on line 681.

Do you think I should still add something here anyways?

Reply:
If it can't happen, then don't have `TryGetImplementedIPredictionTransformer` return a `bool`. Just have it return `Type` and throw inside it if the interface type can't be found.

Reply:
Has this comment been resolved?

Reply:
About to do it. Not sure why codeflow can't find it anymore? I didn't mark it as resolved.

Reply:
It is resolved now. Still not sure why codeflow is having issues showing it.